backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/BaseGitMacheteBranch.java

```diff
@@ -0,0 +1,60 @@
+package com.virtuslab.gitmachete.backend.impl;
+
+import io.vavr.collection.List;
+import io.vavr.control.Option;
+import lombok.Getter;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import com.virtuslab.gitmachete.backend.api.IGitMacheteBranch;
+import com.virtuslab.gitmachete.backend.api.IGitMacheteCommit;
+import com.virtuslab.gitmachete.backend.api.IGitMacheteNonRootBranch;
+import com.virtuslab.gitmachete.backend.api.IGitMacheteRemoteBranch;
+import com.virtuslab.gitmachete.backend.api.SyncToRemoteStatus;
+
+@Getter
+public abstract class BaseGitMacheteBranch implements IGitMacheteBranch {
```

Comment:
Very good job! The many a field (:wink:) repeated in root and non root implementations was a real pain. :sob: